rosidl_generator_c/resource/msg__functions.h.template

```diff
@@ -86,6 +86,66 @@ ROSIDL_GENERATOR_C_PUBLIC_@(spec.base_type.pkg_name)
 void
 @(msg_typename)__destroy(@(msg_typename) * msg);
 
+@# If the message contains an unbounded non-primitive array subfield, declare a resize function
+@{
+unbounded_array_fields = []
+for field in spec.fields:
+    if not field.type.is_array:
+        continue
+    if field.type.is_fixed_size_array():
+        continue
+#    if field.type.is_primitive_type():
+#        continue
+
+    unbounded_array_fields.append(field)
+}@
+
+@[for field in unbounded_array_fields]
+@{
+if field.type.is_primitive_type():
+    subfield_pkg_name = 'rosidl_generator_c'
+    if field.type.type == 'string':
+        subfield_type_name = 'String'
+    else:
+        subfield_type_name = field.type.type
+
+    subfield_array_typename = '%s__%s__Array' % (subfield_pkg_name, subfield_type_name)
+else:
+    subfield_pkg_name = field.type.pkg_name
+    subfield_type_name = field.type.type
+    subfield_array_typename = '%s__msg__%s__Array' % (subfield_pkg_name, subfield_type_name)
+
+subfield_array_decl_guard = '%s__%s__UNBOUNDED_ARRAY__RESIZE_DECL' % (subfield_pkg_name, subfield_type_name)
+}@
+#ifndef @(subfield_array_decl_guard)
+#define @(subfield_array_decl_guard)
+@[if field.type.is_primitive_type()]
+@[  if field.type.type is 'string']
+#include "rosidl_generator_c/string_functions.h"
+@[  else]
+#include "rosidl_generator_c/primitives_array_functions.h"
+@[  end if]
+@[else]
+@{
+subfield_header_file = '%s/msg/%s__functions.h' % (subfield_pkg_name, get_header_filename_from_msg_name(subfield_type_name))
+}@
+#include "@(subfield_header_file)"
+@[end if]
+
+@[if field.type.is_primitive_type()]@
+ROSIDL_GENERATOR_C_PUBLIC
+@[else]
+ROSIDL_GENERATOR_C_PUBLIC_@(subfield_pkg_name)
+@[end if]
+bool
+@(subfield_array_typename)__resize(@(subfield_array_typename) * array, size_t size);
```

Comment:
I very likely just missed it in the "template war", but where are these defined?